### PR TITLE
tex-match: init at 1.2.0

### PR DESCRIPTION
**Pull request by bootstrap-prime**
###### Motivation for this change
I made packages for a thing I use frequently (tex-match) and a fun meme. 

###### Things done
- added self to maintainers
- added "uwuify" package to tools/text
- added tex-match package to tools/typesetting/tex

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by fufexan** [comment]
What you'll want to do is squash the first and last commit you made. So instead of having a separate commit for adding `GithubId`, you'll have both hunks in `c11c0dd`.

File: maintainers/maintainer-list.nix
Comment (fufexan) [inline maintainers/maintainer-list.nix:1271]: Not that it bothers me that much, but usually we try to keep names in lexicographical order. Please move your entry further down the list.

File: pkgs/tools/text/uwuify/default.nix
Comment (fufexan) [inline pkgs/tools/text/uwuify/default.nix:5]: Judging from the project tags, the appropriate version is `0.1.0` https://github.com/Daniel-Liu-c0deb0t/uwu/tree/v0.1.0.

File: pkgs/tools/text/uwuify/default.nix
Comment (fufexan) [inline pkgs/tools/text/uwuify/default.nix:10]: rev can also be changed to `"v${version}"`.

File: pkgs/tools/typesetting/tex/tex-match/default.nix
Comment (fufexan) [inline pkgs/tools/typesetting/tex/tex-match/default.nix:4]: Please keep the same formatting in this file as well (2 spaces indentation).

**Review by fufexan — CHANGES_REQUESTED** [review]

**Comment by fufexan** [comment]
Perhaps I should rephrase - what you will need to do is squash commits together so you end up with 3 commits: one for adding yourself to the maintainers list, one for the uwu package and one for TeX package. The command to do that is `git rebase -i`, but feel free to search "git squash" yourself.

**Comment by bootstrap-prime** [comment]
sorry, I made a mistake with rebase and am trying to fix it

File: pkgs/tools/text/uwuify/default.nix
Comment (fufexan) [inline pkgs/tools/text/uwuify/default.nix:8]: Hm, is there a reason you're not declaring the version directly?

Comment (bootstrap-prime) [reply pkgs/tools/text/uwuify/default.nix:8]: Its used in more than one place (both version and v${}). I declared it this way so I didn't have to make it recursive or write it twice. Is this bad practice?

Comment (fufexan) [reply pkgs/tools/text/uwuify/default.nix:8]: I don't think so. I'm just not used to seeing it. LGTM then.

**Review by fufexan — APPROVED** [review]

File: pkgs/tools/text/uwuify/default.nix
Comment (OPNA2608) [inline pkgs/tools/text/uwuify/default.nix:4]: There's a more recent version available: https://github.com/Daniel-Liu-c0deb0t/uwu/releases/tag/v0.2.1

File: pkgs/tools/text/uwuify/default.nix
Comment (OPNA2608) [inline pkgs/tools/text/uwuify/default.nix:1]: Please avoid using `...` in derivations, we prefer to explicitly specify everything we want to hand through to the derivation. (Exception exist of course)

```suggestion
{ lib, fetchFromGitHub, rustPlatform, stdenv, libiconv }:
```

See below for explanation.

File: pkgs/tools/text/uwuify/default.nix
Comment (OPNA2608) [inline pkgs/tools/text/uwuify/default.nix:18]: Package works on Darwin too, but it needs an extra dependency.

```suggestion
  cargoSha256 = "1v0aldxa31przlq2b6fwnl2grna69gbm33qiaj79cmfng43hyqsy";
  
  buildInputs = lib.optionals stdenv.hostPlatform.isDarwin [ libiconv ];
```

File: pkgs/tools/text/uwuify/default.nix
Comment (OPNA2608) [inline pkgs/tools/text/uwuify/default.nix:25]:
```suggestion
    platforms = platforms.all;
```

File: pkgs/tools/typesetting/tex/tex-match/default.nix
Comment (OPNA2608) [inline pkgs/tools/typesetting/tex/tex-match/default.nix:10]: This doesn't match the version attribute you specified above. Please either change this to

```suggestion
    rev = "v${version}";
```

or change the version attribute to "unstable-" + the date of this commit:
```nix
  version = "unstable-2021-01-12";
```

File: pkgs/tools/text/uwuify/default.nix
Comment (OPNA2608) [inline pkgs/tools/text/uwuify/default.nix:7]: The package name on crates.io and the project README's title are `uwuify`, so that should prolly be the preferred name.
```suggestion
  pname = "uwuify";
```

**Review by OPNA2608 — CHANGES_REQUESTED** [review]
Your commit titles don't follow [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md). Please rename them like so:

1. `maintainers: add bootstrap-prime`
2. `tex-match: init at 1.2.0` (or whatever you end up doing regarding the version - rev mismatch)
3. `uwuify: init at 0.2.1`

**Comment by OPNA2608** [comment]
The suggestion for `libiconv` came from a duplicate PR for uwuify. I'll review both packages on my systems after work and let you know if they work.

You also need to add both packages (in the same commit that contains their respective `default.nix`) to `pkgs/top-level/all-packages.nix` for them to be usable. It's sorted roughly by type of application (comments throughout starting with `###`), then roughly alphabetically within those sections. Example addition:
```nix
  uwuify = callPackage ../tools/text/uwuify { };
```

**Comment by bootstrap-prime** [comment]
I see that a more recent version of uwuify has already been merged. I'll drop that here then and just add myself to the maintainers list and add tex-match.

File: pkgs/tools/typesetting/tex/tex-match/default.nix
Comment (SuperSandro2000) [inline pkgs/tools/typesetting/tex/tex-match/default.nix:8]:
```suggestion
rustPlatform.buildRustPackage {
  pname = "tex-match";
  version = "1.2.0";
```

File: pkgs/tools/typesetting/tex/tex-match/default.nix
Comment (doronbehar) [inline pkgs/tools/typesetting/tex/tex-match/default.nix:15]: Submodules found in the git repo seem related to windows builds. Are you sure they are needed for Darwin / Linux builds?

**Review by doronbehar — COMMENTED** [review]
What a beautiful piece of software!

File: pkgs/tools/typesetting/tex/tex-match/default.nix
Comment (OPNA2608) [reply pkgs/tools/typesetting/tex/tex-match/default.nix:8]: Missing a `rec` there. `let`'d `version` is fine IMO.

Comment (doronbehar) [reply pkgs/tools/typesetting/tex/tex-match/default.nix:8]: > Missing a `rec` there. `let`'d `version` is fine IMO.

It's a matter of personal taste, but indeed `rec` is missing.

**Comment by OPNA2608** [comment]
Package can be compiled & run on Darwin btw, although the actual LaTeX characters wouldn't display (missing dependency at runtime maybe? I don't know). It may also need a different `sha256` on Darwin so maintainability would be pretty *meh*. You can leave it as `platforms.linux` if you don't want to bother with that, even the dev recommends another app for macOS afaict.

File: pkgs/tools/typesetting/tex/tex-match/default.nix
Comment (OPNA2608) [inline pkgs/tools/typesetting/tex/tex-match/default.nix:14]:
```
hash mismatch in fixed-output derivation '/nix/store/sjzbf4hfijz1rm5880dj6b9wnamaaqvw-source':
  wanted: sha256:093wrw4pq1llaivc644r93g96wq5alvh93rdswr6p58zqkqw4zgg
  got:    sha256:1mmiwaks89xxpckbw71b22f08bvnms01jd8vzg11sj8gdfs6w5ps
```
```suggestion
    sha256 = "1mmiwaks89xxpckbw71b22f08bvnms01jd8vzg11sj8gdfs6w5ps";
```

File: pkgs/tools/typesetting/tex/tex-match/default.nix
Comment (OPNA2608) [inline pkgs/tools/typesetting/tex/tex-match/default.nix:22]: (After fixing `sha256`)
```
hash mismatch in fixed-output derivation '/nix/store/cszyrkxl3ckn44c9a0dgq917fqwxng3z-tex-match-1.2.0-vendor.tar.gz':
  wanted: sha256:1ci139q3dzbgsq0pk2hr351v0h0ql2psvqr92iap63apx3v6kqir
  got:    sha256:1sm2fd3dhs59rvmfjzrfz0qwqzyc9dllb8ph0wc2x0r3px16c71x
```
```suggestion
  cargoSha256 = "1sm2fd3dhs59rvmfjzrfz0qwqzyc9dllb8ph0wc2x0r3px16c71x";
```

**Review by OPNA2608 — CHANGES_REQUESTED** [review]

**Review by doronbehar — APPROVED** [review]
The `let version = ; in` thing is an issue of disagreement in Nixpkgs and the majority of people prefer declaring the version along with all other arguments and using `rec`. I too think it's more readable and it's an evaluation price worth paying so that it'd be easier for new contributors to find their way in Nixpkgs, thanks to readability. I will merge if it'll be changed.

**Comment by bootstrap-prime** [comment]
very well. the commit has been modified.

**Comment by siraben** [comment]
I can look into maintaining the darwin side for this package, we need more darwin GUI apps when possible.

**Pull request merged**
